### ISSUE TITLE
`Enter` key in WhatsApp works half the time

### ISSUE THREAD

**Issue by Zwyx** [dessalines/thumb-key#684]
### Steps to reproduce

Simply type a message in WhatsApp.

### Expected behavior

A new line is entered every time the `↵` key is tapped.

### Actual behavior

A new line is entered only half the time the `↵` key is tapped.

### Android version

13

### Device

Google Pixel 4a

### Other details

In the second part of this record, I press `a ↵ a ↵ a ↵ a ↵`, so we should see:

```
a
a
a
a

```

but instead, we see:

```
aa
aa

```

![screen](https://github.com/dessalines/thumb-key/assets/29386932/1b9e56e8-dfad-4791-8a46-e777c019b984)

### Acknowledgements

- [X] I have searched the existing issues and this is a new one, **NOT** a duplicate or related to another open issue.
- [X] This is not a question or a discussion, in which case I should have gone to [lemmy.ml/c/thumbkey](https://lemmy.ml/c/thumbkey)
- [X] This is a **single** bug report,  in case of multiple bugs I will open a separate issue for each one  (they can always link to each other if related)
- [X] I have written a short but informative title.
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements. 🤡
- [X] I have filled out all of the requested information in this form.

**Comment by dessalines** [comment]
Likely means that whatsapp is doing weird things with IME actions. Can you replicate this in any other app?

**Comment by Zwyx** [comment]
No and I've tested quite a few (including Signal, which uses the protocol that WhatsApp uses), but I couldn't see this behaviour anywhere else.

I did notice a different behaviour of the `Enter` key in Slack:
- with MessagEase, the `Enter` key adds a new line in the message you're typing,
- with Thumb-Key, the `Enter` key sends your message! (Slightly unexpected :smile:)

I can create a separate issue for this one if you'd like?

**Comment by dessalines** [comment]
Slack should probably have a setting that make `enter` do a newline rather than an IME send. If not you'll need to wait on #629, which will add a newline hack since discord ignores issues about the same thing.

**Comment by yehoshuapw** [comment]
if it helps, this replicates for me too in Whatsapp (but not in any other place I tested. and I cannot see the keyboard events being sent in adb (if that is even how it is done)). willing to help test anything needed

**Comment by dessalines** [comment]
If this is whats-app specific, then I doubt its a thumb-key issue. I'd open up one over there.
